### PR TITLE
Add ADR-19 "automatic process creation".

### PR DESCRIPTION
**Pull request by jmalloc**
#### What change does this introduce?

This PR introduces ADR-19, describing changes to the process API.
Note that this PR is based on an integration branch that will house all of these changes, and not `main`.

#### Why make this change?

The process API has similar shortcomings to the aggregate API, which were addressed by ADR-16 and ADR-17.
These changes bring the process API more inline with the aggregate API, addressing these shortcomings and improving consistency.

#### Is there anything you are unsure about?

We've made the decision to have a call to `End()` be completely "negated" by a subsequent call to `ExecuteCommand()` or `ScheduleTimeout()`, meaning that any timeout messages that were pending when `End()` was called *remain* pending after handling the message.

#### What issues does this relate to?

#138

### CODE REVIEW

File: docs/adr/0016-automatic-aggregate-creation.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ implement their domain logic in a clear and concise way.
 From the engine implementors perspective it seems that a non-trivial amount of
 validation logic needs to be implemented within the scope to verify that the
 application code is using the scope correctly, even though the engine does not
-really benefit from this requirements.
+really benefit from these requirements.
```
Comment (koden-km) [inline docs/adr/0016-automatic-aggregate-creation.md:44]: Changes to this file seem like a bit of a drive-by...

**Review by koden-km — APPROVED** [review]

File: docs/adr/0019-automatic-process-creation.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Consequences
+
+Largely, this should simplify implementations of both `ProcessMessageHandler` by
+application developers and `Process[Event|Timeout]Scope` by engine developers.
+
+With the removal of `StatelessProcessRoot`, some process implementations will
+require an additional type declaration for their root value.
+
+The removal of `Begin()` does mean that engines longer get an explicit request
```
Comment (danilvpetrov) [inline docs/adr/0019-automatic-process-creation.md:40]:
```suggestion
The removal of `Begin()` does mean that engines get an explicit request longer
```
May be this adverb should be relocated?

**Review by danilvpetrov — APPROVED** [review]

File: docs/adr/0019-automatic-process-creation.md (hunk 2 above)
Comment (koden-km) [reply docs/adr/0019-automatic-process-creation.md:40]: Oh I totally missed that, good catch. I think it's just missing a "no":
```suggestion
The removal of `Begin()` does mean that engines no longer get an explicit request
```

Comment (danilvpetrov) [reply docs/adr/0019-automatic-process-creation.md:40]: Yeah I guess it make more sense this way 👍

File: docs/adr/0016-automatic-aggregate-creation.md (hunk 1 above)
Comment (jmalloc) [reply docs/adr/0016-automatic-aggregate-creation.md:44]: Good call. I originally copied these ADRs to create this one and fixed the same typo in both places, but it only exists here now.

Comment (jmalloc) [reply docs/adr/0016-automatic-aggregate-creation.md]: I've applied these changes to `main` separately.

File: docs/adr/0019-automatic-process-creation.md (hunk 2 above)
Comment (jmalloc) [reply docs/adr/0019-automatic-process-creation.md]: Done!
